src/contrib/dependencyinjection/Akka.DependencyInjection/ServiceProvider.cs

```diff
@@ -0,0 +1,145 @@
+// //     Copyright (C) 2009-2021 Lightbend Inc. <http://www.lightbend.com>
+// //     Copyright (C) 2013-2021 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// // </copyright>
+// //-----------------------------------------------------------------------
+
+using System;
+using Akka.Actor;
+using Akka.Configuration;
+using Akka.Event;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Akka.DependencyInjection
+{
+    /// <summary>
+    /// Provides users with immediate access to the <see cref="IServiceProvider"/> bound to
+    /// this <see cref="ActorSystem"/>, if any.
+    /// </summary>
+    public sealed class ServiceProvider : IExtension
+    {
+        public ServiceProvider(IServiceProvider provider)
+        {
+            Provider = provider;
+        }
+
+        /// <summary>
+        /// The globally scoped <see cref="IServiceProvider"/>.
+        /// </summary>
+        /// <remarks>
+        /// Per https://docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines - please use
+        /// the appropriate <see cref="IServiceScope"/> for your actors and the dependencies they consume. DI is typically
+        /// not used for long-lived, stateful objects such as actors. 
+        ///
+        /// Therefore, injecting transient dependencies via constructors is a bad idea in most cases. You'd be far better off
+        /// creating a local "request scope" each time your actor processes a message that depends on a transient dependency,
+        /// such as a database connection, and disposing that scope once the operation is complete.
+        ///
+        /// Actors are not MVC Controllers. Actors can live forever, have the ability to restart, and are often stateful.
+        /// Be mindful of this as you use this feature or bad things will happen. Akka.NET does not magically manage scopes
+        /// for you.
+        /// </remarks>
+        public IServiceProvider Provider { get; }
+
+        public static ServiceProvider For(ActorSystem actorSystem)
+        {
+            return actorSystem.WithExtension<ServiceProvider, ServiceProviderExtension>();
+        }
+
+        ///// <summary>
+        ///// Uses a delegate to dynamically instantiate an actor where some of the constructor arguments are populated via dependency injection
+        ///// and others are not.
+        ///// </summary>
+        ///// <remarks>
+        ///// YOU ARE RESPONSIBLE FOR MANAGING THE LIFECYCLE OF YOUR OWN DEPENDENCIES. AKKA.NET WILL NOT ATTEMPT TO DO IT FOR YOU.
+        ///// </remarks>
+        ///// <typeparam name="T">The type of actor to instantiate.</typeparam>
+        ///// <param name="producer">The delegate used to create a new instance of your actor type.</param>
+        ///// <returns>A new <see cref="Props"/> instance which uses DI internally.</returns>
+        //public Props Props<T>(Func<IServiceProvider, T> producer) where T : ActorBase
```

Comment:
if I get approval on the API stuff so far, I'll remove this before merging it in